src/nibbles.rs

```diff
@@ -30,8 +30,31 @@ macro_rules! unsafe_unreachable {
 /// the keys in a Merkle Patricia Trie (MPT).
 /// Using nibbles simplifies trie operations and enables consistent key representation in the MPT.
 ///
-/// The internal representation is a [`SmallVec`] that stores one nibble per byte. This means that
-/// each byte has its upper 4 bits set to zero and the lower 4 bits representing the nibble value.
+/// # Internal representation
+///
+/// The internal representation is currently a [`SmallVec`] that stores one nibble per byte. Nibbles
+/// are stored inline (on the stack) up to a length of 64 nibbles, or 32 unpacked bytes. This means
```

Comment:
Bit of a nit pick but I think this should be "32 packed bytes" rather than "32 unpacked bytes"